### PR TITLE
[minor] OSS: bring DDP in the benchmark

### PR DESCRIPTION
**Pull request by blefaudeux**
# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
- [ ] Did you read the [contributor guideline](https://github.com/facebookresearch/fairscale/blob/master/CONTRIBUTING.md)?
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

## What does this PR do?
A suggestion from @msbaines on another PR made me think of that, the benchmark job that we had was not using DDP so not super realistic. The fake pictures were also the same on each rank, which prevents catching some sync issues, fixing that

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by blefaudeux** [comment]
(the values in the benchmark regression test will need to be updated, that's expected. The current commit will fail because of that)

File: benchmarks/oss.py [hunk 1]
```diff
@@ -91,6 +94,7 @@ def train(
         optimizer = ddp.optimizer
         model = ddp
     else:
+        model = DDP(model, device_ids=[rank])
```
Comment (min-xu-ai) [inline benchmarks/oss.py:97]: should we toggle find_unused_parameters to True/False and see its impacts?

**Review by min-xu-ai — APPROVED** [review]
LGTM!

File: benchmarks/oss.py (hunk 1 above)
Comment (blefaudeux) [reply benchmarks/oss.py:97]: ahh, good point ! it was not yet up for review actually, looks like there's something wrong with SDP, looking into that :) I'll test this option

Comment (min-xu-ai) [reply benchmarks/oss.py:97]: oops, sorry, didn't notice that! :-)

**Comment by blefaudeux** [comment]
looks like something is wrong in SDP depending on the world size, and this benchmark change exposed that

**Pull request merged**